### ISSUE TITLE
Modification object during iteration

### ISSUE THREAD

**Issue by zloirock** [ljharb/object.assign#16]
```js
Object.assign({b: 1}, {get a(){delete this.b}, b: 2}).b // => 1
require('object.assign')({b: 1}, {get a(){delete this.b}, b: 2}).b // => undefined (with correct enumeration order)
```
Related [`Object.{values, entries}`](https://bugzilla.mozilla.org/show_bug.cgi?id=1208464#c13)

**Comment by ljharb** [comment]
Spec reference: http://www.ecma-international.org/ecma-262/6.0/#sec-object.assign step 5.c.3.

**Comment by ljharb** [comment]
Fixed - thanks for the report!

**Comment by ljharb** [comment]
Quick revert while I fix it in node 0.8…

**Comment by ljharb** [comment]
k, turns out string properties in node 0.8 and 0.10 are non-enumerable (a bug) and an unrelated test failed as a result of this change. All fixed :-)
